j5/backends/hardware/sr/v4/motor_board.py

```diff
@@ -0,0 +1,190 @@
+"""Hardware Backend for the SR v4 motor board."""
+from functools import wraps
+from typing import Callable, List, Optional, Type, TypeVar, cast
+
+from serial import Serial, SerialException, SerialTimeoutException
+from serial.tools.list_ports import comports
+from serial.tools.list_ports_common import ListPortInfo
+
+from j5.backends import Backend, CommunicationError
+from j5.backends.hardware.env import HardwareEnvironment
+from j5.boards import Board
+from j5.boards.sr.v4.motor_board import MotorBoard
+from j5.components.motor import MotorInterface, MotorSpecialState, MotorState
+
+CMD_RESET = 0
+CMD_VERSION = 1
+CMD_MOTOR = [2, 3]
+CMD_BOOTLOADER = 4
+
+SPEED_COAST = 1
+SPEED_BRAKE = 2
+
+RT = TypeVar("RT")  # pragma: nocover
+
+
+def handle_serial_error(func: Callable[..., RT]) -> Callable[..., RT]:  # type: ignore
+    """
+    Wrap functions that use the serial port, and rethrow the errors.
+
+    This is a decorator that should be used to wrap any functions that call the serial
+    interface. It will catch and rethrow the errors as a CommunicationError, so that it
+    is more explicit what is going wrong.
+    """
+    @wraps(func)
+    def catch_exceptions(*args, **kwargs):  # type: ignore
+        try:
+            return func(*args, **kwargs)
+        except SerialTimeoutException as e:
+            raise CommunicationError(f"Serial Timeout Error: {e}")
```

Comment:
Is it worth raising these `from None`?

Reply:
I'd argue the opposite. If one of the aims of j5 is to empower competitors to explore their kit in depth, hiding error messages from them seems counter intuitive. Also, even if teams are unable technically to dig into an issue this seems like useful information for them to forward on to us if they encounter a weird issue.

I reckon that most of our wrapped errors should be `from e` so that teams get as much information as we can give them about why their robot crashed

Reply:
I have to agree with @sedders123 here. I think that we should aim to expose as much information to competitors as possible to help them debug their problems. The reason for raising a `CommunicationError` here is so that the amount of information does not become overwhelming, there is a simple message at the end of the error stating that the serial interface timed out.

There is scope for a debug mode that raises `from e` and otherwise `from None`, but I think that's a separate issue.